jiant/modules/seq2seq_decoder.py

```diff
@@ -129,6 +138,48 @@ def _initalize_hidden_context_states(self, encoder_outputs, encoder_outputs_mask
 
         return decoder_hidden, decoder_context
 
+    def take_step(
+        self, last_predictions: torch.Tensor, state: Dict[str, torch.Tensor]
+    ) -> Tuple[torch.Tensor, Dict[str, torch.Tensor]]:
+        """Take a decoding step. This is called by the beam search class."""
+
+        decoder_input = self._prepare_decode_step_input(
+            last_predictions,
+            state["decoder_hidden"],
+            state["encoder_outputs"],
+            state["encoder_outputs_mask"],
+        )
+        state["decoder_hidden"], state["decoder_context"] = self._decoder_cell(
+            decoder_input, (state["decoder_hidden"], state["decoder_context"])
+        )
+
+        # output projection
+        proj_input = self._projection_bottleneck(state["decoder_hidden"])
+        # (batch_size, num_classes)
+        output_projections = self._output_projection_layer(proj_input)
+
+        # (batch_size, num_classes)
+        step_logit = output_projections
```

Comment:
what's the point of this line ?

Reply:
Absolutely none, I am not sure why it was there. :)